File: A2OJ/Div2B/508B.py
def swap(l,i,j):
    l = list(l)
    temp = l[i]
    l[i] = l[j]
    l[j] = temp
    p = ""
    for k in l:
        p = p + k
    return p
x = raw_input()
flag = int(x[len(x)-1])
minsofar = -1
l = 0
for i in range(0,len(x)-1):
    if(int(x[i])%2 == 0):
        if(l == 0):
            if(int(x[i]) < flag):
                minsofar= i
                break
            else:
                minsofar = i
            l +=1
        #elif(int(x[i]) <= int(x[minsofar])):
        else:
            l+=1
            if(int(x[i]) < flag):
                minsofar = i
                break
            else:
                minsofar = i
    elif(i == len(x)-2 and minsofar == -1):
        print "-1"
        exit()
print swap(x,len(x)-1,minsofar)

File: Codeforces/Problems/508B.cpp
#include<bits/stdc++.h>
#define append push_back
using namespace std;
int main()
{
	int64_t n;
	cin >> n;
	vector <int> num;
	int64_t m = n;
	while(m>0)
	{
		num.append(m%10);
		m = m /10;
	}
	reverse(num.begin(),num.end());
	int flag = num[num.size()-1];
	int minsofar,l;
	minsofar = -1;
	l = 0;
	for(int i=0;i<num.size()-1;i++)
	{
		if(num[i]%2 == 0)
		{
			if(l==0)
			{
				if(num[i] < flag)
				{
					minsofar = i;
					break;
				}
				else
				{
					minsofar = i;
				}
			}
			else if(num[i] < num[minsofar])
			{
				l++;
				if(num[i] < flag)
				{
					minsofar = i;
					break;
				}
				else
				{
					minsofar = i;
				}
			}
		}
		else if(i == num.size()-2 && minsofar == -1)
		{
			cout << "-1" << endl;
			return 0;
		}
	}
	int temp = num[num.size()-1];
	num[num.size()-1] = num[minsofar];
	num[minsofar] = temp;
	int64_t mew = 0;
	for(int i=0;i<num.size();i++)
	{
		mew = mew*10 + num[i];
	}
	cout << mew << endl;
	return 0;
}





